packages/gator-permissions-snap/src/clients/nonceCaveatClient.ts

```diff
@@ -92,12 +99,12 @@ export class NonceCaveatClient {
       });
     } catch (error) {
       logger.error(`Failed to fetch nonce: ${(error as Error).message}.`);
-      throw new Error('Failed to fetch nonce');
+      throw new ResourceUnavailableError('Failed to fetch nonce');
```

Comment:
Currently we throw a generic error:  

```typescript
throw new ResourceUnavailableError('Failed to fetch nonce');
```

Would it make sense to either:

1. **Surface the original error message**

```typescript
throw new ResourceUnavailableError((error as Error).message);
```

2. **Or include the original error as a cause (where applicable)**

```typescript
throw new ResourceUnavailableError('Failed to fetch nonce', { cause: error });
```

This might help preserve more context for debugging while keeping the high-level signal.

Reply:
I think throwing the cause error to user would not give any additional meaning for the user. The logger should handle the debugging information.